File: 1894-merge-strings-alternately/merge-strings-alternately.java
class Solution {
    public String mergeAlternately(String s, String t) {

        int i = 0;
        int j = 0;
        StringBuilder ans = new StringBuilder();

        while (i < s.length() && j < t.length()) {
            ans.append(s.charAt(i++));
            ans.append(t.charAt(j++));
        }

        while (i < s.length()) {
            ans.append(s.charAt(i++));
        }

        while (j < t.length()) {
            ans.append(t.charAt(j++));
        }

        return ans.toString();
    }
}

File: 1894-merge-strings-alternately/merge-strings-alternately.cpp
class Solution {
public:
    string mergeAlternately(string s, string t) {
        int i = s.length() - 1;
        int j = t.length() - 1;

        string ans = "";
        int k = 0;
        int l = 0;

        while (i >= 0 && j >= 0) {
            ans += s[k++];
            ans += t[l++];

            i--;
            j--;
        }

        if (i >= 0) {
            while (i >= 0) {
                ans += s[k++];
                i--;
            }
        } 
        else {
            while (j >= 0) {
                ans += t[l++];
                j--;
            }
        }

        return ans;
    }
};